components/stateful-form.tsx

```diff
@@ -0,0 +1,207 @@
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { ZodTypeAny, TypeOf } from "zod";
+import { ChangeEvent, useEffect } from "react";
+import {
+  Control,
+  Controller,
+  FieldErrors,
+  FieldValues,
+  Path,
+  UseFormRegister,
+} from "react-hook-form";
+import Phonebox from "./phonebox";
+import Checkbox from "./checkbox";
+import Textbox from "./textbox";
+
+interface StatefulFormProps<Z extends ZodTypeAny> {
+  fields: FormFieldProps[];
+  formValues: TypeOf<Z>;
+  validationSchema: Z;
+  mode?: "onChange" | "onBlur" | "onSubmit";
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  onValidityChange: (e: boolean) => void;
+}
+
+export interface FieldNameProps {
```

Comment:
mm? ...this is very tightly tied to our usecase. please make this stateful form more generic so it can be used for other set of fields not necessarily this